src/invidious/routes/embed.cr

```diff
@@ -201,6 +201,13 @@ module Invidious::Routes::Embed
       return env.redirect url
     end
 
+    if companion_base_url = video.invidious_companion.try &.["baseUrl"].as_s
+      env.response.headers["Content-Security-Policy"] =
+        env.response.headers["Content-Security-Policy"]
+          .gsub("media-src", "media-src #{companion_base_url}")
+          .gsub("connect-src", "connect-src #{companion_base_url}")
+    end
```

Comment:
`video.invidious_companion.try &.["baseUrl"]` gives the companion private_url but without https making requests to the companion fail on an Invidious instance with HTTPS.

![image](https://github.com/user-attachments/assets/8f25b876-4a23-4e17-95e1-ead474d284e0)
![image](https://github.com/user-attachments/assets/87a39917-904f-4fbd-b437-70596e6b0814)

config.yml:
```yaml
invidious_companion:
  - private_url: "http://127.0.0.1:8282"
    public_url: "https://companion.nadeko.net"
```

Reply:
I just noticed that `baseUrl` comes from `[server].base_url` in the companion config file. Sorry!

Reply:
Yes invidious companion needs more doc.